src/datastore/pg-store.ts

```diff
@@ -396,6 +402,76 @@ export class PgStore {
     });
   }
 
+  async getBlocksWithMetadata({ limit, offset }: { limit: number; offset: number }) {
+    return await this.sql.begin(async sql => {
+      // get block list
+      const { results: blocks, total: block_count } = await this.getBlocks({ limit, offset });
+      const blockHashValues: string[] = [];
+      const indexBlockHashValues: string[] = [];
+      blocks.forEach(block => {
+        const indexBytea = block.index_block_hash;
+        const parentBytea = block.parent_index_block_hash;
+        indexBlockHashValues.push(indexBytea, parentBytea);
+        blockHashValues.push(indexBytea);
+      });
+
+      // get txs in those blocks
+      const txQuery = await sql<ContractTxQueryResult[]>`
+        SELECT ${unsafeCols(sql, [...TX_COLUMNS, abiColumn()])}
```

Comment:
You're pulling a bunch of columns here (including the `abi` JOIN) but it seems you only need `tx_id`s in the response (and maybe `index_block_hash` for some comparisons).

You should update this `SELECT` to make this query much faster.